### PR TITLE
Set certificate in some Kestrel tests to avoid global machine state

### PR DESCRIPTION
**Pull request by BrennanConroy**
There were some changes to dev certs which is causing hangs in Kestrel tests in https://github.com/dotnet/aspnetcore/pull/21017. This is because the tests were relying on the dev cert being allowed globally on the test machine, when they should be providing a cert so they don't rely on global state.

Also added a little bit extra test coverage to the existing tests.

### CODE REVIEW

**Comment by Pilchie** [comment]
:eyes:

**Comment by dougbu** [comment]
👁️ have a question: What's the advantage of an 👀 question over an 👀 reaction to the PR description and clicking the `Subscribe` button❔ (I understand just clicking `Subscribe` is invisible and we often want people to know we're looking.)

File: src/Servers/Kestrel/test/BindTests/AddressRegistrationTests.cs [hunk 1]
```diff
@@ -129,12 +129,13 @@ public async Task RegisterAddresses_IPv6_Success(string addressInput, string[] t
...
-        [MemberData(nameof(AddressRegistrationDataIPv6Port5000Default))]
+        [MemberData(nameof(AddressRegistrationDataIPv6Port5000And5001Default))]
         [IPv6SupportedCondition]
         [QuarantinedTest("https://github.com/dotnet/aspnetcore-internal/issues/2711")]
-        public async Task RegisterAddresses_IPv6Port5000Default_Success(string addressInput, string[] testUrls)
+        public async Task RegisterAddresses_IPv6Port5000And5001Default_Success(string addressInput, string[] testUrls)
         {
-            if (!CanBindToEndpoint(IPAddress.Loopback, 5000) || !CanBindToEndpoint(IPAddress.IPv6Loopback, 5000))
+            if ((!CanBindToEndpoint(IPAddress.Loopback, 5000) || !CanBindToEndpoint(IPAddress.IPv6Loopback, 5000)) &&
+                (!CanBindToEndpoint(IPAddress.Loopback, 5001) || !CanBindToEndpoint(IPAddress.IPv6Loopback, 5001)))
```
Comment (dougbu) [inline src/Servers/Kestrel/test/BindTests/AddressRegistrationTests.cs:138]: When an agent is in this state, should the test fail and output information about what's holding the port?

**Comment by BrennanConroy** [comment]
Putting 👀 (or any comment) will add you to the mentions in the email metadata, so you can easily filter issues that you're actually interested in. Kevin told me once that he was subscribed to everything and I am too, I know that I have filters based on the email metadata so adding those kinds of comments are super useful for getting updates.

**Review by dougbu — APPROVED** [review]
Should have said :shipit:

Unfortunately https://netcorenativeassets.blob.core.windows.net seems to have been flaky during the build.  See also #20577 though a retry will be needed here.

**Review by halter73 — APPROVED** [review]

File: src/Servers/Kestrel/test/BindTests/AddressRegistrationTests.cs (hunk 1 above)
Comment (halter73) [reply src/Servers/Kestrel/test/BindTests/AddressRegistrationTests.cs:138]: No. It just passes. Every AddressRegistrationTest that test the ability to bind to a specific port (as opposed to "port 0") has done this for years now.

**Comment by Pilchie** [comment]
Yep, @BrennanConroy nailed why I use :eyes:.  I get email for anything I'm "participating", but see the rest of my notifications in the GitHub notifications UI.

**Comment by javiercn** [comment]
We probably want to port this back to 3.1 too

**Comment by dougbu** [comment]
> We probably want to port this back to 3.1 too

No disagreement here. It's a test-only change and can therefore be merged whenever the branch is open. @BrennanConroy❔

Same thing for d6409371aeec aka #21493 @javiercn❔

**Comment by BrennanConroy** [comment]
Sure, I'll make a PR for this one

**Comment by javiercn** [comment]
> Same thing for [d640937](https://github.com/dotnet/aspnetcore/commit/d6409371aeeca6df7e126b1e0c51410544fc2869) aka #21493 @javiercn❔

I already have a PR out for it https://github.com/dotnet/aspnetcore/pull/21496

**Pull request merged**